doc/release-notes/template.sgml

```diff
@@ -0,0 +1,117 @@
+<!doctype linuxdoc system>
+<article>
+<title>Squid @SQUID_VERSION@ release notes</title>
```

Comment:
According to the new Makefile rule (introduced in this PR), this template file name (and the file names of future vN release notes made using this template file) should end with `sgml.in`, not `.sgml`.

Reply:
This is the template from which the release-N.sgml.in files are created on the major version branching (eg today for v6-v7).

Reply:
This is the parent template from which the `release-N.sgml.in` templates are creates.

Reply:
> This is the parent template from which the `release-N.sgml.in` templates are created on major version branching.

The above known fact underscores my change request.

Reply:
@kinkie, to break the tie, what do you think this filename should be?

Reply:
> @kinkie, to break the tie, what do you think this filename should be?

I believe we can make a technically correct decision here without resorting to a popularity contest. Templates (and files that are copied to produce templates) that require `.sgml.in`-to-`.sgml` processing should have `.sgml.in` extension. No argument has been presented against applying that (very natural!) principle AFAICT.

Reply:
There is a (small but present) risk that if we name it .sgml.in and someone were to type "make template.sgml" it would trigger the global %.in rules in Makefiles, so I'm going to lean on keeping it named template.sgml . It's clear enough and few enough people are touching this that the benefits in consistency are not as important

Reply:
> There is a (small but present) risk that if we name it .sgml.in and someone were to type "make template.sgml" it would trigger the global %.in rules in Makefiles

What are we risking? Why making template.sgml is so bad that we are willing to misname a file to avoid that risk? I would actually argue the opposite -- making template.sgml from template.sgml.in is a useful feature, not a problem we need to mitigate! I can use that feature to compare the edited release notes with their "virgin" copy, for example (without unexpanded macros in template.sgml.in showing up all over the diff).

Also, are you ignoring the "risk" of of someone typing "make template.html" and creating _that_ file from the misnamed template.sgml? If so, why ignore _that_ risk?! In fact, unlike SGML generation, HTML generation is a lot more invasive because it overwrites `$(top_builddir)/RELEASENOTES.html`. AFAICT, this ought to be viewed as a much "riskier" problem that we should avoid as a bigger evil among the two choices.

Reply:
My understanding of the process from v6 onwards is that:
- when vX is branched, release-v(X+1).sgml is created copying from template.sgml
- release-vX.sgml and release-v(X+1).sgml get updated as part of the regular PR landing

So.. template.sgml really is only a template, and is only ever used once.

A question (for v7+, so out of scope here) is whether we want to switch to friendlier format, e.g. markdown, which would offer better readability and tooling

Reply:
> release-v(X+1).sgml is created copying from template.sgml

If that statement is accurate, then my change request should be resolved without any changes.

My understanding is that the above statement is inaccurate. Amos's [response](https://github.com/squid-cache/squid/pull/1264#discussion_r1096869010) and code (discussed below) confirm my understanding AFAICT.

AFAICT, in [reality](https://github.com/squid-cache/ci/blob/686087cca9bbdae3e9d00f3fe7abda0bde6ce559/release/mk-release-series.sh#L48), `release-v(X+1).sgml.in` (**not** `release-v(X+1).sgml`!) is "created by copying from template.sgml". Does this correction allow you to see why I am requesting this renaming change? In summary, this PR adds a template file that is used to create `.in` files (that require .sgml.in->.sgml preprocessing before they actually become SGML) but misnames the file as if it were pure SGML file (that contains text that does not require such preprocessing).

Here is the relevant [code](https://github.com/squid-cache/ci/blob/686087cca9bbdae3e9d00f3fe7abda0bde6ce559/release/mk-release-series.sh#L48):

```shell
cp doc/release-notes/template.sgml doc/release-notes/release-${NEWVER}.sgml.in
```

> release-vX.sgml and release-v(X+1).sgml get updated as part of the regular PR landing

AFAICT, we will be updating `release-vX.sgml.in` files, not `.sgml` files. Again, here is the relevant [code](https://github.com/squid-cache/ci/blob/686087cca9bbdae3e9d00f3fe7abda0bde6ce559/release/mk-release-series.sh#L49):

```shell
git add doc/release-notes/release-${NEWVER}.sgml.in
```

Reply:
@kinkie, it would be nice to get your updated reaction on this change request as well. I hope my response changes your opinion of what the right file name is here because your earlier feedback was probably based on false assumptions. If you say nothing, the wrong file name will be officially merged.

Reply:
This change request has not been addressed, but fixing the filename is not critical. @yadij, if this PR needs to be merged urgently, I do not object to you removing my request for @kinkie review (and re-clearing this PR for merging). Otherwise, I hope Francesco will find the time to revisit this soon.

Reply:
I'm happy to keep as is, pending a reassessment of sgml as format for the future.

Reply:
FWIW, this change request is independent from this file current SGML format -- markdown format would have the same naming problem, just spelled with different letters. AFAICT, responses so far were not about the problem that this change request is actually about.